### PR TITLE
Added an isData optional parameter to hs.websocket:send()

### LINKED ISSUE #2752: hs.websocket sends all messages as WebSocket Binary even when the message is a lua string
If any of you are using ProPresenter, you probably know it uses a simple websocket interface for remote control.

`hs.websocket` used to be able to talk to it, but it doesn't seem to be working anymore.

I've done a fair amount of testing, and it seems like `hs.websocket` is working fine with every websocket server I've tested it with, but it is not working with ProPresenter for some reason. My guess is that hammerspoon is sending something different in the headers that causes ProPresenter to ignore the data.

Another option is that hammerspoon is having a problem accessing the ProPresenter websocket endpoint. ProPresenter serves web sockets over a subdirectory. Here's the url for my case:

`ws://192.168.10.50:60157/stagedisplay`

Can anyone confirm this problem? Is there a way for me to inspect or modify the websocket headers sent from hammerspoon?

EDIT:
See below... I've determined the problem seems to be that `hs.websocket:send` always sends websocket messages as WebSocket Binary (opcode set to 0x0010) instead of Text (opcode set to 0x0001). Some programs (like ProPresenter) expect Text messages and only listen to text messages.

### PR DESCRIPTION
**Pull request by latenitefilms**
- Closes #2752

### CODE REVIEW

**Comment by latenitefilms** [comment]
@asmagill - Are you able to idiot check this?

@cmsj - I saw your message on the Google Groups, but pod headers we are using look different to the ones you linked to (i.e. there is no `sendString` and `sendData` methods).

**Review by asmagill — APPROVED** [review]
lgtm

**Comment by cmsj** [comment]
> @cmsj - I saw your message on the Google Groups, but pod headers we are using look different to the ones you linked to (i.e. there is no `sendString` and `sendData` methods).

Ah, looks like that was added in the (now archived) GitHub repo for SocketRocket, but never released.

File: extensions/websocket/internal.m [hunk 1]
```diff
@@ -163,21 +163,35 @@ static int websocket_new(lua_State *L) {
 
 /// hs.websocket:send(message) -> object
```
Comment (cmsj) [inline extensions/websocket/internal.m:164]: Needs a [, isData]

**Review by cmsj — CHANGES_REQUESTED** [review]

**Review by cmsj — APPROVED** [review]

**Pull request merged**